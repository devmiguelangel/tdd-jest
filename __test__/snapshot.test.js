import { getCharacter } from '../snapshot';
import rick from '../rick.json';

describe('Snapshot test', () => {
  test('Snapshot', () => {
    expect(getCharacter(rick)).toMatchSnapshot();
  });

  test('Snapshot failure', () => {
    const user = {
      createdAt: new Date(),
      id: Math.floor(Math.random() * 20),
    };

    expect(user).toMatchSnapshot();
  });

  test('Snapshot exception', () => {
    const user = {
      id: Math.floor(Math.random() * 20),
      name: "Miguel Angel",
    };

    expect(user).toMatchSnapshot({
      id: expect.any(Number),
    });
  });

});
