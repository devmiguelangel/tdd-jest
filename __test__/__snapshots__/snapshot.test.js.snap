// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Snapshot test Snapshot exception 1`] = `
Object {
  "id": Any<Number>,
  "name": "Miguel Angel",
}
`;

exports[`Snapshot test Snapshot failure 1`] = `
Object {
  "createdAt": 2019-11-26T22:08:13.710Z,
  "id": 5,
}
`;
